Sporadic probe unit test failure

<!-- Please only use this template for submitting reports about failing tests in iotex-core CI jobs -->

**Which jobs are failing**:

https://circleci.com/gh/iotexproject/iotex-core/6125

**Which test(s) are failing**:

```
=== RUN   TestBasicProbe
--- FAIL: TestBasicProbe (0.00s)
    require.go:794: 
        	Error Trace:	probe_test.go:26
        	            				probe_test.go:50
        	Error:      	Received unexpected error:
        	            	Get http://localhost:7788/liveness: dial tcp 127.0.0.1:7788: connect: connection refused
        	Test:       	TestBasicProbe
=== RUN   TestReadniessHandler
2019-04-29T07:32:09.641Z	ERROR	probe/probe.go:74	Failed to listen on probe port	{"error": "listen tcp :7788: bind: address already in use"}
github.com/iotexproject/iotex-core/pkg/probe.(*Server).Start.func1
	/go/src/github.com/iotexproject/iotex-core/pkg/probe/probe.go:74
--- FAIL: TestReadniessHandler (0.00s)
    probe_test.go:27: 
        	Error Trace:	probe_test.go:27
        	            				probe_test.go:99
        	Error:      	Not equal: 
        	            	expected: 202
        	            	actual  : 503
        	Test:       	TestReadniessHandler
    probe_test.go:27: 
        	Error Trace:	probe_test.go:27
        	            				probe_test.go:99
        	Error:      	Not equal: 
        	            	expected: 202
        	            	actual  : 503
        	Test:       	TestReadniessHandler
FAIL
coverage: 85.3% of statements
FAIL	github.com/iotexproject/iotex-core/pkg/probe	0.009s
Exited with code 1
```

**Since when has it been failing**:

**Testgrid link**:

**Reason for failure**:

**Anything else we need to know**:

--- a/e2etest/local_transfer_test.go
+++ b/e2etest/local_transfer_test.go
@@ -279,6 +279,7 @@ func TestLocalTransfer(t *testing.T) {
 	// Start server
 	go itx.StartServer(context.Background(), svr, probeSvr, cfg)
 	defer func() {
+		require.Nil(probeSvr.Stop(ctx))
 		require.Nil(svr.Stop(ctx))
 	}()
 

